pkg/api/v1beta2/dynakube/properties.go

```diff
@@ -432,6 +432,10 @@ func (dk *DynaKube) MetaDataEnrichmentEnabled() bool {
 	return dk.Spec.MetadataEnrichment.Enabled
 }
 
+func (dk *DynaKube) MetadataEnrichmentNamespaceSelector() *metav1.LabelSelector {
+	return &dk.Spec.MetadataEnrichment.NamespaceSelector
+}
```

Comment:
do we have to return a pointer? what do we gain by that?

we can easily check if its unset/empty with `.Size()`

Reply:
(if we use a pointer for the `OneAgentNamespaceSelector` then we are dumb, and should fix that at some point 😛 )

Reply:
Indeed we return a pointer. for OneAgentNamespaceSelector. Should I update this within this PR? Wdyt?

Reply:
you can try, 
BUT if it introduces way too many changes then just leave it, lets not blow up the PR
- We can create a ticket about it, that can be more general, and I think we might already have one, related to using the kubebuilder "marker-comments" more, we could do it there

Reply:
It seems to me as though we can not call `Size()` on `MetadataEnrichmentSelector` if we do not return a pointer. Can you confirm?

Reply:
you can't use it directly, because `.Size()` was written for a pointer receiver.

usually Golang hides this fact, and auto converts from "value" to "pointer", but it can't do it with the return value of a function.

so incase of:
```go

func (dk *DynaKube) OneAgentNamespaceSelector() metav1.LabelSelector { ... }
```
```go
dk.OneAgentNamespaceSelector().Size() // not allowed

nl := dk.OneAgentNamespaceSelector()
nl.Size() // allowed
```

Reply:
but don't worry about it, we should fix this an other time, because I am flip-flopping about which should be the better solution

Reply:
(reason for flip-flopping:
We store the `LabelSelector` in the CR as a "value" so `MetadataEnrichmentNamespaceSelector()` will **never** return `nil`, as an empty `LabelSelector`'s pointer is not nil.
And if its nicer to use, then maybe its better this way 😅 
)

Reply:
Thanks for elaboration. Then I will let it as it is here and maybe we will create stories to fix this in the future. cc @chrismuellner